Enterprise-CMCS/cmcs-eregulations
# EREGCSC-2156 -- Refined policy repository results styling

Resolves [EREGCSC-2156](https://jiraent.cms.gov/browse/EREGCSC-2156)

**Description:**

Refines the results styling on the front end for `/policy-repository` documents.

Refactors RelatedSections component and related python logic to better route users to the correct section in the reader view.

Adds Cypress end to end tests for authenticated section of `/policy-repository`.  Uses `cypress.env.json` variables locally (which each dev will need to create, since they will not be committed to source control).  Uses seeded `dev` user for experimental deployments.  Uses newly created `test-user`, which has no privileges, for `dev`/`val`/`prod` deployments.

**This pull request changes:**

- Fixes bug in `reader.py` that was not properly getting latest version date and redirecting to correct location for certain URL structures, specifically when linking to a Subpart without a date parameter in the URL
- Simplifies `last-updated-date` logic for SPA footer
- Refactors `RelatedSections.vue` to not need as many passed-in props to create a URL to the related section.  Also simplifies associated utility filter method `locationUrl`
   - Moves location filtering logic to this component, which encapsulates it in a single location.  Filtering had previously been happening in each front end SPA view (Resources.vue, Search.vue, PolicyRepository.vue)
   - Uses Vue's [provide/inject ](https://vuejs.org/guide/components/provide-inject.html) to reduce prop drilling for non-reactive props
   - Moves RelatedSections styles into their own SCSS partial stylesheet
- Adds styles related to Policy Repository items
- Adds Cypress e2e tests for basic testing of Policy Repository authentication flow

**Steps to manually verify this change:**

1. Visit most recent experimental deployment action for this PR and ensure all tests have passed.
2. Visit experimental deployment and ensure that "Mock Dental, Dentures" has a style Document Type, Subjects, and Locations
3. Feel free to update other documents to add Document Type, Subjects, and Locations to see them
4. If you'd like to test locally, build the project using `make local` and create the admin instance using `make local.createadmin`.  Create a test user
5. use the `cypress.env.json.template` file to create your own `cypress.env.json` config file and fill in the test user and test password values
6. Run `make test.cypress` and `make test.vitest` to make sure all tests pass
